### PR TITLE
[e2e] Fix object compare

### PR DESCRIPTION
**Pull request by axinging**
The predict results of most models are objects, instead of arrays. Current implementation assumes that the predict results are arrays, which results in some fake pass. For example, when tint changed the shared memory limits, most models should fail the conformance test. However, these cases are passed when we compare these results as arrays.

BUG: https://github.com/tensorflow/tfjs/issues/5397

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/5384)

### CODE REVIEW

**Comment by axinging** [comment]
@qjia7 @haoyunfeix @xhcao @gyagp, PTAL

File: e2e/benchmarks/local-benchmark/util.js [hunk 1]
```diff
@@ -66,3 +66,102 @@ function queryTimerIsEnabled() {
...
+    }
+  }
+}
+
+function expectArraysClose(actual, expected, epsilon, key) {
+  if (epsilon == null) {
+    epsilon = tf.test_util.testEpsilon();
+  }
+
+  if (key == 'data') {
```
Comment (qjia7) [inline e2e/benchmarks/local-benchmark/util.js:156]: Can we make sure that if key is `data`, the value must be `0|1`?

**Review by qjia7 — COMMENTED** [review]
Good finding, Xing.
Would like @pyu10055 @lina128 @jinjingforever take a look, thanks.

File: e2e/benchmarks/local-benchmark/util.js (hunk 1 above)
Comment (axinging) [reply e2e/benchmarks/local-benchmark/util.js:156]: This path is specially for bodypix, which has a special member data in its predict results:
`export type SemanticPersonSegmentation = {
  data: Uint8Array,
  width: number,
  height: number,
  allPoses: Pose[],
};`
The current value for this data member is 0|1. But I am not sure if the same rule applies for the other models.

**Review by lina128 — COMMENTED** [review]
Thank you, Xing, this is great!

> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/5384#-:-MfnCd9ZALxANdxOD_QH:b-46vdxe)* status: 0 of 1 approvals obtained (waiting on @axinging, @jinjingforever, @pyu10055, and @qjia7)

___
*[e2e/benchmarks/local-benchmark/util.js, line 102 at r1](https://reviewable.io/reviews/tensorflow/tfjs/5384#-MfnAmzx2_bEcSWthKAD:-MfnAmzx2_bEcSWthKAE:b-bxfquy) ([raw file](https://github.com/tensorflow/tfjs/blob/1fc1899b53c246c8fe50d12e5c830074db1fdaac/e2e/benchmarks/local-benchmark/util.js#L102)):*
<details><summary><i alt="summary">Quoted 14 lines of code…</i></summary>

> ```
>     if (isArray) {
>       expectArraysClose(actual[key], expected[key], epsilon, key);
>     } else if (!isObject && (!predicate(actual[key], expected[key]))) {
>       throw new Error(`Objects differ: actual[${key}] = ${
>           JSON.stringify(actual[key])}, expected[${key}] = ${
>           JSON.stringify(expected[key])}!`);
>     } else if (
>         isObject &&
>         !expectObjectsPredicate(
>             actual[key], expected[key], epsilon, predicate)) {
>       throw new Error(`Objects differ: actual[${key}] = ${
>           JSON.stringify(actual[key])}, expected[${key}] = ${
>           JSON.stringify(expected[key])}!`);
>     }
> ```
</details>

Can we refactor the conditions to be more readable?
```
if (isArray) {
  expectArraysClose(...);
} else if(isObject) {
  expectObjectsPredicate(...); // if you want to log extra context for error, surround it with a try ... catch ...
} else {
  predicate(...);
}
```

**Comment by axinging** [comment]
@lina128 , Updated, PTAL

**Review by lina128 — APPROVED** [review]
Thank you so much, great stuff! LGTM.

> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/5384#-:-MfpmYMA4xgYG9J9iqmF:b-s8x2m5)* status: :shipit: complete! 1 of 1 approvals obtained (waiting on @axinging, @jinjingforever, @pyu10055, and @qjia7)

**Pull request merged**